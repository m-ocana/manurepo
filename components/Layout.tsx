import Head from "next/head";
import Link from "next/link";
import Header from "./Header";
import BackToHome from "./BackToHome";
import utilStyles from "../styles/utils.module.css";
import styled from "styled-components";
import { TITLE } from "../globals/site";

const Container = styled.div`
  max-width: 36rem;
  padding: 0 1rem;
  margin: 3rem auto 6rem;
`;

export default function Layout({
  children,
  home,
}: {
  children: React.ReactNode;
  home?: boolean;
}) {
  return (
    <Container>
      <Head>
        <link rel="icon" href="/favicon.ico" />
        <meta
          name="description"
          content="Collection of cheatsheets, snippets and utils"
        />
        <meta name="og:title" content={TITLE} />
      </Head>
      <Header home={home} />
      <main>{children}</main>
      {!home && <BackToHome />}
    </Container>
  );
}
