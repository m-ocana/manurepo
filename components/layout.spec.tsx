import React from "react";
import renderer from "react-test-renderer";
import Layout from "./layout";

describe("Layout", () => {
  it("scaffolds the home page", () => {
    const tree = renderer.create(<Layout home children={null} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("scaffolds any other page", () => {
    const tree = renderer.create(<Layout children={null} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
