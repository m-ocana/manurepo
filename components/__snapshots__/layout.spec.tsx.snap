// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout scaffolds any other page 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <a>
      <img
        alt="The Manurepo"
        className="headerImage"
        src="/images/repo.svg"
      />
    </a>
    <h2
      className="headingLg"
    >
      <a
        className="colorInherit"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        The Manurepo
      </a>
    </h2>
  </header>
  <main />
  <div
    className="backToHome"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      ‚Üê Back to home
    </a>
  </div>
</div>
`;

exports[`Layout scaffolds the home page 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <img
      alt="The Manurepo"
      className="headerHomeImage"
      src="/images/repo.svg"
    />
    <h1
      className="heading2Xl"
    >
      The Manurepo
    </h1>
  </header>
  <main />
</div>
`;
